options.go

```diff
@@ -286,7 +286,7 @@ type PushOptions struct {
 	// ForceWithLease allows a force push as long as the remote ref adheres to a "lease"
 	ForceWithLease *ForceWithLease
 	// PushOptions sets options to be transferred to the server during push.
-	Options map[string]string
+	Options []string
```

Comment:
Similarly, a map/kv pair here would make it more straight-forward for handling keys that must overwrite options from the config.

Reply:
Duplicates should be allowed. In case of no options are given, we should derive them from config files.

Reply:
My point was regarding [this](https://git-scm.com/docs/git-push#Documentation/git-push.txt-pushpushOption):

> This is a multi-valued variable, and an empty value can be used in a higher priority configuration file (e.g. .git/config in a repository) to clear the values inherited from a lower priority configuration files (e.g. $HOME/.gitconfig).

However, that is orthogonal to the actual transport-level behaviour, so happy for us to transition the type here and handle the config options separately.